encoder.py

```diff
@@ -0,0 +1,106 @@
+    :param pool_strategy: the pooling strategy. Options are:
+        - `None`: Means that the output of the model will be the 4D tensor
+            output of the last convolutional block.
+        - `mean`: Means that global average pooling will be applied to the
+            output of the last convolutional block, and thus the output of
+            the model will be a 2D tensor.
+        - `max`: Means that global max pooling will be applied.
+    :param channel_axis: The axis of the color channel, default is 1
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+    DEFAULT_TRAVERSAL_PATH = ['r']
+
+    def __init__(
+        self,
+        model_name: str = 'mobilenet_v2',
+        pool_strategy: str = 'mean',
+        channel_axis: int = 1,
+        load_pre_trained_from_path: Optional[str] = None,
+        default_traversal_path: Optional[List[str]] = None,
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.default_traversal_path = self.DEFAULT_TRAVERSAL_PATH if default_traversal_path is None\
+            else default_traversal_path
+
+        self.channel_axis = channel_axis
+        # axis 0 is the batch
+        self._default_channel_axis = 1
+        self.model_name = model_name
+        if pool_strategy not in ('mean', 'max'):
+            raise NotImplementedError(f'unknown pool_strategy: {self.pool_strategy}')
+        self.pool_strategy = pool_strategy
+
+        if load_pre_trained_from_path:
+            # set env var as described in https://pytorch.org/vision/stable/models.html
+            model = getattr(models, self.model_name)(pretrained=False)
+            model.load_state_dict(torch.load(load_pre_trained_from_path))
+
+
+        model = getattr(models, self.model_name)(pretrained=True)
+        self.model = model.features.eval()
+        self.model.to(torch.device('cpu'))
+        if self.pool_strategy is not None:
+            self.pool_fn = getattr(np, self.pool_strategy)
+
+    def _get_features(self, content):
+        return self.model(content)
+
+    def _get_pooling(self, feature_map: 'np.ndarray') -> 'np.ndarray':
+        if feature_map.ndim == 2 or self.pool_strategy is None:
+            return feature_map
+        return self.pool_fn(feature_map, axis=(2, 3))
+
+    @requests
+    # TODO: per request traversal path
+    # TODO: add batching
+    def encode(self, docs: DocumentArray, **kwargs):
+        chunks = DocumentArray(
+            docs.traverse_flat(self.default_traversal_path)
+        )
+        images = np.stack(chunks.get_attributes('blob'))
+        images = self._maybe_move_channel_axis(images)
+
+        _input = torch.from_numpy(images)
+        features = self._get_features(_input).detach()
+        features = self._get_pooling(features.numpy())
+
+        for doc, embed in zip(chunks, features):
+            doc.embedding = embed
+
+        return chunks
```

Comment:
I don't think return chunks is a good way. add embeddings should be enough. This will replace the original docs

Reply:
Changed it to return nothing.